### ISSUE TITLE
Anti AdBlock KIller: Does It Work on Userscript?

### ISSUE THREAD

**Issue by malicerainbow** [quoid/userscripts#386]
_System Information:_
> macOS or iOS version: iPhone 7 Plus, iOS 15.7.1
> Userscripts version: 1.3.3
> Safari version: 15.7.1
> Is this issue related to script injection?: I'm not sure, but I've tested a more complex script called "YouTube Return Dislikes" which did work on the Userscript app. When I try this script though, it didn't. I need to narrow down what the issue is (script, phone, app, etc.).
> Did the test script (pasted above) successfully run on your machine?: The test script didn't work, but the Userscript app isn't allowing me to paste scripts.

**(Here the script for Anti AdBlock Killer. This is what I'm trying to use and I need to confirm it works for Userscript. I know it's functional since I had a friend get Tampermonkey, and they ran it with AdBlock Plus to get the filter.)**
https://raw.githubusercontent.com/reek/anti-adblock-killer/master/anti-adblock-killer.user.js

**Comment by malicerainbow** [comment]
It was written with safari compatibility but so far through YouTube videos only noticeable of safari on Mac computer it has not been seen if it works for safari of iOS 

Also of YouTube videos it’s shown to also work on userscript on mac or windows but not of userscript of iOS

**Comment by malicerainbow** [comment]
Okay so sorry I mistaken the userscript Mac YouTube videos of return YouTube dislike with anti ad blocker killer. So far there is no YouTube videos of anti adblocker killer with userscript only tampermonkey. But I had a friend test it on their computer with tampermonkey and Adblock plus and the script worked functionally so the script is not the problem. I’m not sure if the problem is userscript the app. Or my phone or AdGuard adblocker. 

That’s why I asked if you could check to see if the script would work on your userscript app? My phone is old so there’s a chance it’s because of my phone but I won’t know unless someone with a newer iOS tries the script on their userscript app and my friend doesn’t want to install the app


But just to confirm the script works! And here is even a YouTube video to show that it’s visibly workable with tampermonkey 


https://m.youtube.com/watch?v=oH7FVbowIpo


Now my friend is willing to test if it would work on their safari of Mac computer of userscript should we try that?

**Comment by malicerainbow** [comment]
Yes and it saids one of its supportive browsers is safari but I don’t know if that’s safari Mac computer safari iOS or both 
![Uploading EA2C3A61-FFA4-4600-8FFB-2F57724D83AB.jpeg…]()

**Comment by malicerainbow** [comment]
![087BF343-6CD5-42E4-8AAE-34389B2A8F9E](https://user-images.githubusercontent.com/118444664/203655360-b036251b-f7c8-44a0-8e37-e14017a7853a.jpeg)

**Comment by malicerainbow** [comment]
Oh I thought you were talking about safari I didn’t know you meant userscript my bad!

**Comment by malicerainbow** [comment]
But wait isnt userscript equivalent to tampermonkey?

**Comment by malicerainbow** [comment]
I've now had a friend test both the Return YouTube Dislikes script and the one that goes with that Anti AdBlock Killer test site, using both mobile (iOS) and computer (macOS), and it worked with both. On both, using Safari, he used the Userscripts app and the AdGuard AdBlock App. All that now being known, that the AntI AdBlock Killer script will work with the Userscripts script manager, why isn't it working for me?
<img width="1434" alt="Screen Shot 2022-11-29 at 11 19 16 PM" src="https://user-images.githubusercontent.com/118444664/204708888-713d5c77-cc2d-46dd-859b-602354215982.png">
<img width="294" alt="Screen Shot 2022-11-29 at 11 19 37 PM" src="https://user-images.githubusercontent.com/118444664/204708916-e12efaf2-004f-47ab-8a7f-1010421bff03.png">

**Comment by malicerainbow** [comment]
But the extension aka the app is the exact same app on the Mac App Store if you look at the screenshots we sent we showed visually the script is working with UserScript

the website would be saying please remove adblocker if it wasn’t 

don’t you see in the screenshot that the script is working with userscript by Justin Wasack 

The script is working!!! We confirmed this with Userscript on MacOS 


But the script is not working on Userscript app for mobile iOS

**Comment by malicerainbow** [comment]
If it’s not supported by the userscript app than why was it supported on the userscript app of macOS by same developer?

**Comment by malicerainbow** [comment]
We literally show you on the screenshots how we evaluated it working 
![5481E537-B124-4551-BD18-B24FA86565DB](https://user-images.githubusercontent.com/118444664/204764598-d37b38cf-9712-462b-83c5-f32dfe0cc5dc.jpeg)
![B5D58362-4425-4AD7-8B6C-9B38610F0FEF](https://user-images.githubusercontent.com/118444664/204764610-aad6d6d9-f021-46a2-a835-a49a2774639d.jpeg)



If you look carefully that is the userscript icon with the red little circle with number 1 letting you know the script is working!!!!
![BF987FDE-1F73-4DE0-9922-54D39DEBD764](https://user-images.githubusercontent.com/118444664/204765097-178ba2a3-be70-4597-8d6d-b56327556729.png)


This is what happens if you don’t have the script it shows this 

so the script works! 


I don’t understand how you don’t see how we evaluated 

it literally shows you on the screenshot I sent that the script is active on userscript and working!!!

so why is it working on macOS userscript but not iOS userscript?

it’s the exact app by the exact developer Justin Wasack 

it’s the exact “extension” aka app

**Comment by malicerainbow** [comment]
It’s literally not allowing the website to detect my adblocker what more do you need?

**Comment by malicerainbow** [comment]
What do you need me to do ACTCD in order to confirm that it’s working properly on UserScripts?

I really don’t get this the scripts language is to command AdGuard to not detectable by any website it blocks ads from 


The script running on Userscripts of macOS worked where the website could not detect AdGuard 

so what more do you need!!!!!!!!!!!!!

I really don’t get what else you need to prove the script is compatible with Userscripts by Justin Wasack

**Comment by malicerainbow** [comment]
I was not saying the red 1 mark indicates it working I was saying the fact the website is not detecting the adblocker indates that it’s working on Userscripts

**Comment by malicerainbow** [comment]
I know the red 1 mark just informs you that only one script is running for the particular site your on


And that since it did not lock my friend out of saying please remove adblocker 


That means the script is working!!!!!!!!!!!

**Comment by malicerainbow** [comment]
I saw the script working with my own eyes the website did not detect an adblocker 


See it for yourself go to this link with an advlocker on and you will see it will not allow you to continue viewing the site until you turn off your adblocker!!!!!!

http://antiblock.org/?p=v3&demo


But with script it will make the website not detect the adblocker which worked for my friend!!!!

**Comment by malicerainbow** [comment]
Can we not use the visual confirmation that the site is allowing him to view it without removing his adblocker as indication that the script is working on Userscripts!!! 


I mean seriously the scripts language is to command AdGuard through the custom filter that the script works along side with to not be detectable on any website AdGuard is running on blocking ads 


Ad as you can see from the screenshot the website did not detect the adblocker 
![0CCF8FF2-3E3E-4362-8DCB-5C3707F39E07](https://user-images.githubusercontent.com/118444664/204935568-310db409-4877-48eb-b2ed-075fbde69cf0.jpeg)

that is what the website looks like if it does not detect an adblocker 



the script works properly on Userscripts!!!!

**Comment by quoid** [comment]
@malicerainbow It sounds like you are saying the script is working properly, if so, that's great. 

Can you use [the discussions tab](https://github.com/quoid/userscripts/discussions/categories/general) for further comments about whether or not scripts are working?

**Comment by malicerainbow** [comment]
But quoid my issue is it’s working properly on Userscripts MacOS but not Userscripts IOS and I’d like to know why if I can please?

it’s the exact same app of iOS by the exact same developer 

Yet it’s only working on macOS not iOS for this script

**Comment by malicerainbow** [comment]
But quoid my issue is it’s working properly on Userscripts MacOS but not Userscripts IOS and I’d like to know why if I can please?

it’s the exact same app of iOS by the exact same developer

Yet it’s only working on macOS not iOS for this script

**Comment by malicerainbow** [comment]
Mr Justin your script manager of macOS is working with the script but your script manager of iOS is not!

**Comment by quoid** [comment]
@malicerainbow 

Does the script rely on the AdGuard extension or something other AdBlock extension to function properly?

You seem to reference this within your comment, can you validate that?

**Comment by malicerainbow** [comment]
The script will work with Adblock, Adblock plus, uorigin, and AdGuard adblocker Mr Justin sir 


The reason I say AdGuard is because only AdGuard will allow custom filters on iOS  to ad the scripts filter 

I assure you thou the issue is not with AdGuard 

and the script does work with your script manager on macOS but not iOS
